src/ShaderTools/Hlsl/Binding/Binder.Declarations.cs

```diff
@@ -267,7 +267,7 @@ private BoundNode BindFunctionDefinition(FunctionDefinitionSyntax declaration, S
                 Bind(declaration.Semantic, BindVariableQualifier);
 
             var functionBinder = (functionOwner != null && functionOwner.Kind == SymbolKind.Class)
```

Comment:
But I thought you removed `ClassSymbol`, which is the only time `SymbolKind.Class` was created? And you added `IsClass` on the syntax node.